File: KotlinMobileApp/Java/Start/app/src/main/java/com/rachelleboyette/catquiz/adapters/CatQuizViewAdapter.java
package com.rachelleboyette.catquiz.adapters;

import android.content.Context;
import android.content.Intent;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Button;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.rachelleboyette.catquiz.R;
import com.rachelleboyette.catquiz.common.Constants;
import com.rachelleboyette.catquiz.customviews.CatQuizCompletionView;
import com.rachelleboyette.catquiz.enums.ChoiceStage;
import com.rachelleboyette.catquiz.models.CatQuizModel;
import com.rachelleboyette.catquiz.models.CatQuizModelStore;
import com.rachelleboyette.catquiz.views.CatQuizPageActivity;

import java.util.List;

public class CatQuizViewAdapter extends RecyclerView.Adapter<CatQuizViewAdapter.ViewHolder> {

    private List<CatQuizModel> catQuizModels;
    public CatQuizViewAdapter(List<CatQuizModel> catQuizModelList) {
        this.catQuizModels = catQuizModelList;
    }

    @NonNull
    @Override
    public ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        return new ViewHolder(LayoutInflater.from(parent.getContext()).inflate(R.layout.cat_quiz_listview_adapter, parent));
    }

    @Override
    public void onBindViewHolder(@NonNull ViewHolder holder, int position) {
        holder.bind(catQuizModels.get(position));
    }

    @Override
    public int getItemCount() {
        return catQuizModels.size();
    }

    class ViewHolder extends RecyclerView.ViewHolder implements View.OnClickListener {

        private CatQuizModel catQuizModel;

        public ViewHolder(@NonNull View itemView) {
            super(itemView);
        }

        @Override
        public void onClick(View view) {
            Context context = view.getContext();

            Intent intent = new Intent(context, CatQuizPageActivity.class);
            intent.putExtra(Constants.questions, CatQuizModelStore.getQuestions())
                    .putExtra(Constants.choice, CatQuizModelStore.getQuizChoices())
                    .putExtra(Constants.quizType, catQuizModel.getQuizType())
                    .putExtra(Constants.stage, ChoiceStage.FIRST);
            context.startActivity(intent);
        }

        public void bind(CatQuizModel catQuizModel) {
            this.catQuizModel = catQuizModel;

            bindCompletionRate(this.catQuizModel, itemView);
            bindQuizType(this.catQuizModel, itemView);
        }

        private void bindQuizType(CatQuizModel catQuizModel, View itemView) {
            Button button = itemView.findViewById(R.id.quizButton);
            switch(catQuizModel.getQuizType()) {
                case GENERAL:
                    button.setText(R.string.general_quiz);
                    break;
                case FOOD:
                    button.setText(R.string.food_quiz);
                    break;
                default:
                    break;
            }
        }

        private void bindCompletionRate(CatQuizModel catQuizModel, View itemView) {
            CatQuizCompletionView completionView = itemView.findViewById(R.id.catImageView);
            completionView.setCompletionRate(catQuizModel.getCompletionRate());
        }
    }
}

File: KotlinMobileApp/KotlinMobileApp/app/src/main/java/com/neudesic/catquiz/adapters/CatQuizViewAdapter.kt
package com.neudesic.catquiz.adapters

import android.content.Intent
import android.view.View
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView
import com.neudesic.catquiz.R
import com.neudesic.catquiz.common.Constants
import com.neudesic.catquiz.enums.ChoiceStage
import com.neudesic.catquiz.enums.QuizType
import com.neudesic.catquiz.helpers.inflate
import com.neudesic.catquiz.models.CatQuizModel
import com.neudesic.catquiz.models.CatQuizModelStore
import com.neudesic.catquiz.views.CatQuizPageActivity
import kotlinx.android.synthetic.main.cat_quiz_listview_adapter.view.*

class CatQuizViewAdapter(private val catQuizzes: List<CatQuizModel>) : RecyclerView.Adapter<CatQuizViewAdapter.ViewHolder>() {

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        return ViewHolder(parent.inflate(R.layout.cat_quiz_listview_adapter))
    }

    override fun getItemCount() = catQuizzes.size

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        holder.bind(catQuizzes[position])
    }

    class ViewHolder(itemView: View): RecyclerView.ViewHolder(itemView), View.OnClickListener
    {
        private lateinit var catQuiz: CatQuizModel

        //TODO change this to the catQuiz stage vs setting manual stage
        override fun onClick(view: View?) {
            val context = view?.context
            val intent = Intent(context, CatQuizPageActivity::class.java)

            intent.putExtra(Constants.questions, CatQuizModelStore.getQuestions())
                    .putExtra(Constants.choice, CatQuizModelStore.getQuizChoices())
                    .putExtra(Constants.quizType, catQuiz.quizType)
                    .putExtra(Constants.stage, ChoiceStage.FIRST)

            context?.startActivity(intent)
        }

        init {
            itemView.setOnClickListener(this)
        }

        fun bind(catQuiz: CatQuizModel) {
            this.catQuiz = catQuiz

            bindCompletionRate(this.catQuiz, itemView)
            bindQuizTypeText(this.catQuiz, itemView)
        }

        private fun bindQuizTypeText(catQuiz: CatQuizModel, itemView: View) {
            when(this.catQuiz.quizType) {
                QuizType.GENERAL -> itemView.quizButton.setText(R.string.general_quiz)
                QuizType.FOOD -> itemView.quizButton.setText(R.string.food_quiz)
            }
        }

        private fun bindCompletionRate(catQuiz: CatQuizModel, itemView: View) {
            itemView.catImageView.completionRate = catQuiz.completionRate
        }
    }
}